### ISSUE TITLE
cannot import name hostapd_controller

### ISSUE THREAD

**Issue by eloelo22** [wifiphisher/wifiphisher#887]
Before opening a new issue, please make sure that what you are about to report has not been reported already. If not, use the following template for reporting.

**Version**: 1.4

**Description**: i ran install.py everything wen well then after execute wifiphisher it returns an error

**Script output**: 
Traceback (most recent call last):
  File "/usr/local/bin/wifiphisher", line 11, in <module>
    load_entry_point('wifiphisher==1.4', 'console_scripts', 'wifiphisher')()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 587, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2800, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2431, in load
    return self.resolve()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2437, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/pywifiphisher.py", line 27, in <module>
    import wifiphisher.common.accesspoint as accesspoint
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/common/accesspoint.py", line 7, in <module>
    from roguehostapd import hostapd_controller
ImportError: cannot import name hostapd_controller

**Comment by anakin1028** [comment]
Hi thanks for reporting. Once #880 is merged, this issue should be resolved.

**Comment by eloelo22** [comment]
thank you

**Comment by Learpcs** [comment]
I have the same problem too

**Script:** 

```
Traceback (most recent call last):
  File "/usr/local/bin/wifiphisher", line 9, in <module>
    load_entry_point('wifiphisher==1.4', 'console_scripts', 'wifiphisher')()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 542, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2570, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2230, in load
    return self.resolve()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2236, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/pywifiphisher.py", line 27, in <module>
    import wifiphisher.common.accesspoint as accesspoint
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/common/accesspoint.py", line 7, in <module>
    from roguehostapd import hostapd_controller
ImportError: cannot import name hostapd_controller

```

**Comment by anakin1028** [comment]
@eloelo22  @Learpcs The issue should be fixed with newest version now.

**Comment by hearthacker1** [comment]
the import module are missing and the wifiphisher are going to install .please resolve this problem
 Traceback (most recent call last):
  File "./wifiphisher", line 16, in <module>
    
  File "/root/wifiphisher/wifiphisher/pywifiphisher.py", line 25, in <module>
    import wifiphisher.common.interfaces as interfaces
  File "/root/wifiphisher/wifiphisher/common/interfaces.py", line 11, in <module>
    import dbus
ImportError: No module named dbus

**Comment by fritz276** [comment]
Is that the same bug? Worked for me untill the merges some days ago.

Traceback (most recent call last):
  File "bin/wifiphisher", line 16, in <module>
    from wifiphisher.pywifiphisher import run
  File "/home/ps/wifiphisher/wifiphisher/pywifiphisher.py", line 27, in <module>
    import wifiphisher.common.accesspoint as accesspoint
  File "/home/ps/wifiphisher/wifiphisher/common/accesspoint.py", line 8, in <module>
    import roguehostapd.config.hostapdconfig as hostapdconfig
ImportError: No module named config.hostapdconfig

**Comment by anakin1028** [comment]
Can you rerun the setup.py of wifiphisher?

**Comment by fritz276** [comment]
I did that more than once :-)

**Comment by fritz276** [comment]
Thanks! That did the trick!

**Comment by datteroandrea** [comment]
I am having this issue:
root@stecco-K30AD-M31AD-M51AD:/home/stecco/Scaricati/wifiphisher-master/bin# wifiphisher 
Traceback (most recent call last):
  File "/usr/local/bin/wifiphisher", line 9, in <module>
    load_entry_point('wifiphisher==1.4', 'console_scripts', 'wifiphisher')()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 542, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2569, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2229, in load
    return self.resolve()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2235, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/pywifiphisher.py", line 25, in <module>
    import wifiphisher.common.interfaces as interfaces
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/common/interfaces.py", line 11, in <module>
    import dbus
ImportError: No module named dbus
can you help me solve this please?

**Comment by anakin1028** [comment]
Can you try the following:
```
apt-get remove python-dbus
pip install python-dbus
```

**Comment by hearthacker1** [comment]
thanks for giving me a reply to solve this problem .but i now i have a
problem of
python scripts like this
Traceback (most recent call last):
  File "./wifiphisher", line 16, in <module>

  File "/root/wifiphisher/wifiphisher/pywifiphisher.py", line 20, in
<module>
    import wifiphisher.common.extensions as extensions
  File "/root/wifiphisher/wifiphisher/common/extensions.py", line 11, in
<module>
    import scapy.layers.dot11 as dot11
  File "/usr/local/lib/python2.7/dist-packages/scapy/layers/dot11.py", line
12, in <module>
    from scapy.packet import *
  File "/usr/local/lib/python2.7/dist-packages/scapy/packet.py", line 39
    class Packet(BasePacket, metaclass = Packet_metaclass):
                                       ^
SyntaxError: invalid syntax
please tell me how to resolve this issue to run the wifiphisher
i am waiting for your respones

**Comment by anakin1028** [comment]
Can you try the same methods:
```
apt-get remove scapy
pip install scapy
```

**Comment by hearthacker1** [comment]
yes i do but they show me this error
root@khalid:~/wifiphisher# pip install scapy
Collecting scapy
  Downloading scapy-2.3.3.tgz (1.4MB)
    24% |███████▉                        | 348kB 7.7kB/s eta
0:02:19Exception:
Traceback (most recent call last):
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/_vendor/urllib3/response.py",
line 302, in _error_catcher
    yield
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/_vendor/urllib3/response.py",
line 384, in read
    data = self._fp.read(amt)
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/_vendor/cachecontrol/filewrapper.py",
line 60, in read
    data = self.__fp.read(amt)
  File "/usr/lib/python3.6/http/client.py", line 449, in read
    n = self.readinto(b)
  File "/usr/lib/python3.6/http/client.py", line 493, in readinto
    n = self.fp.readinto(b)
  File "/usr/lib/python3.6/socket.py", line 586, in readinto
    return self._sock.recv_into(b)
  File "/usr/lib/python3.6/ssl.py", line 1009, in recv_into
    return self.read(nbytes, buffer)
  File "/usr/lib/python3.6/ssl.py", line 871, in read
    return self._sslobj.read(len, buffer)
  File "/usr/lib/python3.6/ssl.py", line 631, in read
    v = self._sslobj.read(len, buffer)
socket.timeout: The read operation timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/basecommand.py",
line 215, in main
    status = self.run(options, args)
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/commands/install.py",
line 324, in run
    requirement_set.prepare_files(finder)
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/req/req_set.py",
line 380, in prepare_files
    ignore_dependencies=self.ignore_dependencies))
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/req/req_set.py",
line 620, in _prepare_file
    session=self.session, hashes=hashes)
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/download.py",
line 821, in unpack_url
    hashes=hashes
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/download.py",
line 659, in unpack_http_url
    hashes)
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/download.py",
line 882, in _download_http_url
    _download_url(resp, link, content_file, hashes)
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/download.py",
line 603, in _download_url
    hashes.check_against_chunks(downloaded_chunks)
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/utils/hashes.py",
line 46, in check_against_chunks
    for chunk in chunks:
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/download.py",
line 571, in written_chunks
    for chunk in chunks:
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/utils/ui.py",
line 139, in iter
    for x in it:
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/download.py",
line 560, in resp_read
    decode_content=False):
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/_vendor/urllib3/response.py",
line 436, in stream
    data = self.read(amt=amt, decode_content=decode_content)
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/_vendor/urllib3/response.py",
line 401, in read
    raise IncompleteRead(self._fp_bytes_read, self.length_remaining)
  File "/usr/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File
"/usr/local/lib/python3.6/dist-packages/pip-9.0.3-py3.6.egg/pip/_vendor/urllib3/response.py",
line 307, in _error_catcher
    raise ReadTimeoutError(self._pool, None, 'Read timed out.')
pip._vendor.urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='
pypi.python.org', port=443): Read timed out.
 please help me again

**Comment by hearthacker1** [comment]
i am waiting for response . Teach me a best trick to learn this hacking
i am a begginer in kali linux

**Comment by usama7628674** [comment]
@hearthacker1 It seems that pip is pointing to python3.6.Can you do this
`sudo python -m pip install -U --force-reinstall pip`
Then re-run the above commands.

**Comment by hearthacker1** [comment]
thanks for giving again a replay but i can't resolve this issue but typing
the commond

sudo python -m pip install -U --force-reinstall pip
and they show me the error of
oot@khalid:~/wifiphisher/bin# ./wifiphisher
Traceback (most recent call last):
  File "./wifiphisher", line 16, in <module>

  File "/root/wifiphisher/wifiphisher/pywifiphisher.py", line 20, in
<module>
    import wifiphisher.common.extensions as extensions
  File "/root/wifiphisher/wifiphisher/common/extensions.py", line 11, in
<module>
    import scapy.layers.dot11 as dot11
  File "/usr/local/lib/python2.7/dist-packages/scapy/layers/dot11.py", line
12, in <module>
    from scapy.packet import *
  File "/usr/local/lib/python2.7/dist-packages/scapy/packet.py", line 39
    class Packet(BasePacket, metaclass = Packet_metaclass):
                                       ^
SyntaxError: invalid syntax

help me once again usama . i am waiting for your response again

**Comment by usama7628674** [comment]
@hearthacker1 `apt-get remove scapy`
`pip install scapy`

**Comment by hearthacker1** [comment]
i did these steps but can't resolve this problem . the invalid syntex are
still comeing
root@khalid:~/wifiphisher/bin# pip install scapy
Requirement already satisfied: scapy in /usr/lib/python2.7/dist-packages
root@khalid:~/wifiphisher/bin# ./wifiphisher
Traceback (most recent call last):
  File "./wifiphisher", line 16, in <module>

  File "/root/wifiphisher/wifiphisher/pywifiphisher.py", line 20, in
<module>
    import wifiphisher.common.extensions as extensions
  File "/root/wifiphisher/wifiphisher/common/extensions.py", line 11, in
<module>
    import scapy.layers.dot11 as dot11
  File "/usr/local/lib/python2.7/dist-packages/scapy/layers/dot11.py", line
12, in <module>
    from scapy.packet import *
  File "/usr/local/lib/python2.7/dist-packages/scapy/packet.py", line 39
    class Packet(BasePacket, metaclass = Packet_metaclass):
                                       ^
SyntaxError: invalid syntax

**Comment by usama7628674** [comment]
@hearthacker1 `pip uninstall scapy`
`pip install scapy --no-cache-dir`

**Comment by hearthacker1** [comment]
oh again showing this bloody error can't resolve this problem
Collecting scapy
  Downloading scapy-2.3.3.tgz (1.4MB)
    100% |████████████████████████████████| 1.4MB 2.1kB/s
Installing collected packages: scapy
  Running setup.py install for scapy ... done
Successfully installed scapy-2.3.3
root@khalid:~/wifiphisher/bin# ./wifiphisher
Traceback (most recent call last):
  File "./wifiphisher", line 16, in <module>

  File "/root/wifiphisher/wifiphisher/pywifiphisher.py", line 20, in
<module>
    import wifiphisher.common.extensions as extensions
  File "/root/wifiphisher/wifiphisher/common/extensions.py", line 11, in
<module>
    import scapy.layers.dot11 as dot11
  File "/usr/local/lib/python2.7/dist-packages/scapy/layers/dot11.py", line
12, in <module>
    from scapy.packet import *
  File "/usr/local/lib/python2.7/dist-packages/scapy/packet.py", line 39
    class Packet(BasePacket, metaclass = Packet_metaclass):
                                       ^
SyntaxError: invalid syntax
sorry but help me

**Comment by usama7628674** [comment]
@hearthacker1 Just write `scapy` in terminal and see i it starts or not.

**Comment by anakin1028** [comment]
Someone has the same problem in #753, however I don't know the root cause. You still can have a try with command:
```
apt-get remove python-scapy
```
I'm thinking the system default install a version of scapy with Python3.

**Comment by hearthacker1** [comment]
I try this and they work but here is a new problem come in starting of
wifiphisher
wifiphisher
root@khalid:~/wifiphisher/bin# ./wifiphisher
[*] Starting Wifiphisher 1.4GIT ( https://wifiphisher.org ) at 2018-03-25
18:48
[!] Failed to find an interface with AP mode
[+] Show your support!
[+] Follow us: https://twitter.com/wifiphisher
[+] Like us: https://www.facebook.com/Wifiphisher
[+] Captured credentials:
[!] Closing

please also resolive this error again for me
thanks for giving replay

**Comment by anakin1028** [comment]
Ok this means your adaptor doesn't support Access Point mode. Wifiphisher will require the adaptor to support AP mode to lunch a fake Access Point.

**Comment by datteroandrea** [comment]
Ehi anakin1028 i tried what you have said and when i do apt-get remove python-dbus it says that the package is not installed but when i say pip install python-dbus it says:"could not find a version satisfies the requirment python-dbus" i have tried installing dbus via apt-get it does install it but the tool is not working.

**Comment by hearthacker1** [comment]
Hi bro i hope you are fine and thanks for all replays .
Now tell me which one adopter i buy from market in pakistan

**Comment by hearthacker1** [comment]
can you tell me another methord to hack wifi

**Comment by Mr-BlackHat** [comment]
@hearthacker1 there are lots of YouTube videos available on how to decrypt password from a pcap file u can follow them step by step aircrack-ng is the tool that can help u or use Fluxsion to fool the owner of the wifi to enter password for fake AP

**Comment by hearthacker1** [comment]
i am try fluxsion but still can't give any success to me .

**Comment by Mr-BlackHat** [comment]
@hearthacker1 contact me on skype or give me ur id

**Comment by hearthacker1** [comment]
so you want to talk with me ya

**Comment by hearthacker1** [comment]
send me your contact number i make call to you

**Comment by hearthacker1** [comment]
i am using the whatsapp id .
here is my id name shafiq mushfiq
id number is 03068017328
i am waiting for you call please contact me

**Comment by mhomami** [comment]
**Version*** 1.4
**Description** Tried the above fixes for "cannot import name hostapd_controller" on laptop installation and USB Live installation. Both continue to issue the same error. I've uninstalled and reinstalled rogueapd.

**Script Output**

Traceback (most recent call last):
  File "/usr/local/bin/wifiphisher", line 11, in <module>
    load_entry_point('wifiphisher==1.4', 'console_scripts', 'wifiphisher')()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 572, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2752, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2405, in load
    return self.resolve()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2411, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/pywifiphisher.py", line 27, in <module>
    import wifiphisher.common.accesspoint as accesspoint
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/common/accesspoint.py", line 7, in <module>
    from roguehostapd import hostapd_controller
ImportError: cannot import name hostapd_controller

**Comment by krushiraj** [comment]
@anakin1028 tried what you said above but still it gives the same ImportError.

**Comment by mhomami** [comment]
Same here. I downloaded the #880 merge and the program now launches but won't deauth clients.

**Comment by mhomami** [comment]
Installed the 1.4 release again on a fresh install of Kali after update/dist-upgrade. Same error. I also uninstalled and reinstalled both hostadp and roguehostadp and ran setup.py again. No joy.

To test the theory, I downloaded the #902 branch this time and managed to get it working first time. The Master appears to be missing the fix implemented in #880 or a new issue has popped up.

**Comment by mhomami** [comment]
Scratch my last. On attack, it exits telling me roguehostapd is not installed. I've uninstalled and reinstalled with the pip command and downloaded the zip version and tried installed it through there too. No joy.

I tried downloaddng #880 to see if I can get that version to work. Still exits with error roguehostapd not installed in system.

**Comment by mhomami** [comment]
Figured it out. Install roguehostapd from following instructions:

git clone https://github.com/wifiphisher/roguehostapd.git # Download the latest version
cd roguehostapd # Switch to the roguehostapd directory
python setup.py install # Build the shared library of hostapd

pip install roguehostapd installed version 1.2.3 which, for whatever reason, is not working with 1.4.

I must have done this last night on my other system too and forgot about it.

**Comment by mhomami** [comment]
1.4 Master still broken:

Traceback (most recent call last):
  File "/usr/local/bin/wifiphisher", line 11, in <module>
    load_entry_point('wifiphisher==1.4', 'console_scripts', 'wifiphisher')()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 480, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2693, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2324, in load
    return self.resolve()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2330, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/pywifiphisher.py", line 27, in <module>
    import wifiphisher.common.accesspoint as accesspoint
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/common/accesspoint.py", line 7, in <module>
    from roguehostapd import hostapd_controller
ImportError: cannot import name hostapd_controller
root@kali:~/roguehostapd# hostapd -v

hostapd v2.6 installed. 

Any ideas?

**Comment by anakin1028** [comment]
From the log you are still using the old version of roguehostapd. Can you do the following?

```
pip uninstall roguehostapd
pip install roguehostapd --no-cache-dir
```

**Comment by mhomami** [comment]
Did this several times already. Repeated just now, results:

root@kali:~# pip uninstall roguehostapd
Uninstalling roguehostapd-1.1.2:
  /usr/local/lib/python2.7/dist-packages/roguehostapd-1.1.2-py2.7-linux-x86_64.egg
Proceed (y/n)? y
  Successfully uninstalled roguehostapd-1.1.2
root@kali:~# pip uninstall roguehostapd
Uninstalling roguehostapd-1.1.2:
  /usr/local/lib/python2.7/dist-packages/roguehostapd-1.1.2-py2.7.egg
Proceed (y/n)? y
  Successfully uninstalled roguehostapd-1.1.2
root@kali:~# pip uninstall roguehostapd
Cannot uninstall requirement roguehostapd, not installed
root@kali:~# pip install roguehostapd --no-cache-dir
Collecting roguehostapd
  Downloading https://files.pythonhosted.org/packages/ca/0e/67991a382656c30ec4f6197db8e96bf8d3b87cc0093254f0c97340fa7c53/roguehostapd-1.2.3.tar.gz (1.9MB)
    100% |████████████████████████████████| 1.9MB 1.8MB/s 
Installing collected packages: roguehostapd
  Running setup.py install for roguehostapd ... done
Successfully installed roguehostapd-1.2.3
root@kali:~# wifiphisher
Traceback (most recent call last):
  File "/usr/local/bin/wifiphisher", line 11, in <module>
    load_entry_point('wifiphisher==1.4', 'console_scripts', 'wifiphisher')()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 480, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2693, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2324, in load
    return self.resolve()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2330, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/pywifiphisher.py", line 27, in <module>
    import wifiphisher.common.accesspoint as accesspoint
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/common/accesspoint.py", line 7, in <module>
    from roguehostapd import hostapd_controller
ImportError: cannot import name hostapd_controller

**Comment by anakin1028** [comment]
I know what's the problem. You have to clone the newest version of wifiphisher. The 1.4 version used the outdated version of roguehostapd.

**Comment by GHOST07v** [comment]
hi, i have this problem, any help please?
Traceback (most recent call last):
  File "/usr/local/bin/wifiphisher", line 9, in <module>
    load_entry_point('wifiphisher==1.4', 'console_scripts', 'wifiphisher')()
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/pywifiphisher.py", line 693, in run
    engine.start()
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/pywifiphisher.py", line 390, in start
    self.opmode.initialize(args)
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/common/opmode.py", line 53, in initialize
    interfaces.is_add_vif_required(args)
  File "/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/common/interfaces.py", line 798, in is_add_vif_required
    card = pyw.getcard(vif)
  File "/usr/local/lib/python2.7/dist-packages/PyRIC-0.1.6.3-py2.7.egg/pyric/pyw.py", line 269, in getcard
    if nlsock is None: return _nlstub_(getcard, dev)
  File "/usr/local/lib/python2.7/dist-packages/PyRIC-0.1.6.3-py2.7.egg/pyric/pyw.py", line 2236, in _nlstub_
    return fct(*argv)
  File "/usr/local/lib/python2.7/dist-packages/PyRIC-0.1.6.3-py2.7.egg/pyric/pyw.py", line 270, in getcard
    return devinfo(dev, nlsock)['card']
  File "/usr/local/lib/python2.7/dist-packages/PyRIC-0.1.6.3-py2.7.egg/pyric/pyw.py", line 1057, in devinfo
    'mac': _hex2mac_(nl.nla_find(rmsg, nl80211h.NL80211_ATTR_MAC)),
  File "/usr/local/lib/python2.7/dist-packages/PyRIC-0.1.6.3-py2.7.egg/pyric/pyw.py", line 1821, in _hex2mac_
    return ":".join(['{0:02x}'.format(c) for c in v])
TypeError: 'NoneType' object is not iterable

**Comment by blackHatMonkey** [comment]
@GHOST07v Your issue seems to be reported in #955.
